### PR TITLE
treewide: remove torch and related packages

### PR DESCRIPTION
**Pull request by bhipple**
See #71888 for details.

CC @NixOS/nixos-release-managers @nh2 @teto

ZHF: #80379

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg eval

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23714,7 +23699,7 @@ in
   multimc = libsForQt5.callPackage ../games/multimc { };
 
   inherit (callPackages ../games/minetest {
-	inherit (darwin) libiconv;
+    inherit (darwin) libiconv;
```
Comment (bhipple) [inline pkgs/top-level/all-packages.nix]: Unrelated change, my `emacs` replaced the tab character someone inserted into this file with the correct number of spaces.

File: pkgs/development/libraries/torch/default.nix [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
-  version = "0.0pre20160820";
-  pname = "torch";
-  buildInputs = [
-    luajit openblas imagemagick cmake curl fftw gnuplot unzip qt4
-    libjpeg zeromq3 ncurses openssl libpng readline pkgconfig
-    zlib libX11 which
-  ];
-
-  src = fetchgit {
-    url = "https://github.com/torch/distro";
```
Comment (bhipple) [inline pkgs/development/libraries/torch/default.nix]: We're running version `0.0pre20160820`, which hasn't built in a long time, and upstream https://github.com/torch/distro/blob/master/README.md says, as of 3 years ago:

> NOTE: Torch is not actively developed anymore and is in maintenance mode.

**Comment by nh2** [comment]
> @nh2 

Btw I know nothing about torch, on #83567 I just cherry-picked @teto's cleanup.

@bhipple on https://github.com/NixOS/nixpkgs/pull/83567#issuecomment-605382445 @worldofpeace said:

> We really should have a `throw` in `aliases.nix`. Would love to have a `throws.nix` eventually.

Do you know how to do that? If yes, could you do it for these packages?

**Comment by bhipple** [comment]
I think the idea is to check in something in the top-level `aliases.nix`, like this:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix#L51

So that if an unsuspecting user goes to try to do something like `nix-build -A torch`, they will get the alias telling them it was deleted rather than an attribute not found error.

Is that right @worldofpeace?  In this particular case I think we'd be fine with or without the aliases, since it's been broken for a long time and upstream died out years ago by the looks of it, so there probably won't be many users trying it out.

**Comment by worldofpeace** [comment]
> I think the idea is to check in something in the top-level `aliases.nix`, like this:
> https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix#L51
> 
> So that if an unsuspecting user goes to try to do something like `nix-build -A torch`, they will get the alias telling them it was deleted rather than an attribute not found error.
> 
> Is that right @worldofpeace? In this particular case I think we'd be fine with or without the aliases, since it's been broken for a long time and upstream died out years ago by the looks of it, so there probably won't be many users trying it out.

Yes, that is spot on. Considering it has died out for years a silent removal could be ok. But in a general sense, the silent removal of anything usually upsets end users.
